### PR TITLE
Add Client Side Public Key Authentication

### PR DESCRIPTION
**Pull request by ejohnstown**

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -6697,11 +6702,441 @@ int SendServiceAccept(WOLFSSH* ssh, byte serviceId)
...
+        word32 idx = 0;
+        ret = wc_EccPrivateKeyDecode(authData->sf.publicKey.privateKey,
+                &idx, &keySig->ks.ecc.key,
+                authData->sf.publicKey.privateKeySz);
+    }
+
+    if (ret == WS_SUCCESS) {
+        if (authData->sf.publicKey.hasSignature) {
+            int sigSz = wc_ecc_sig_size(&keySig->ks.ecc.key);
+            sigSz = 64;
```
Comment (cconlon) [inline src/internal.c]: Should sigSz be forced to 64 here, when the line above it sets it based on wc_ecc_sig_size()?

File: src/internal.c [hunk 2]
```diff
@@ -6697,11 +6702,441 @@ int SendServiceAccept(WOLFSSH* ssh, byte serviceId)
...
+{
+    wc_HashAlg hash;
+    byte digest[WC_MAX_DIGEST_SIZE];
+    word32 digestSz;
+    word32 begin;
+    enum wc_HashType hashId = WC_HASH_TYPE_SHA;
+    int ret = WS_SUCCESS;
+    byte* r;
+    byte* s;
+    byte sig[512];
```
Comment (cconlon) [inline src/internal.c]: Should MAX_ECC_BITS be used here instead of 512?

File: src/internal.c (hunk 1 above)
Comment (ejohnstown) [reply src/internal.c]: Oh yeah, that was for something I was testing and forgot to remove. Thanks for catching that.

File: src/internal.c (hunk 2 above)
Comment (ejohnstown) [reply src/internal.c]: It should be big enough to fit a signature for a prime 521 key, so 144 bytes. I think.

Comment (ejohnstown) [reply src/internal.c]: Reduced the sig buffer to 72 bytes, which is the size of a signature for a prime256 key.

**Pull request merged**